src/array/cuda/spmat_op_impl.cu

```diff
@@ -0,0 +1,255 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file array/cuda/spmat_op_impl.cu
+ * \brief Sparse matrix operator CPU implementation
+ */
+#include <dgl/array.h>
+#include <vector>
+#include <unordered_set>
+#include <numeric>
+#include "../../runtime/cuda/cuda_common.h"
+#include "../../cuda_utils.h"
+
+namespace dgl {
+
+using runtime::NDArray;
+
+namespace aten {
+namespace impl {
+
+///////////////////////////// CSRIsNonZero /////////////////////////////
+
+/*!
+ * \brief Search adjacency list linearly for each (row, col) pair and
+ * write the matched position in the indices array to the output.
+ * 
+ * If there is no match, -1 is written.
+ * If there are multiple matches, only the first match is written.
+ */
+template <typename IdType>
+__global__ void _LinearSearchKernel(
+    const IdType* indptr, const IdType* indices,
+    const IdType* row, const IdType* col,
+    int64_t row_stride, int64_t col_stride,
+    int64_t length, IdType* out) {
+  int tx = blockIdx.x * blockDim.x + threadIdx.x;
+  int stride_x = gridDim.x * blockDim.x;
+  int rpos = tx, cpos = tx;
+  while (tx < length) {
+    out[tx] = -1;
+    const IdType r = row[rpos], c = col[cpos];
+    for (IdType i = indptr[r]; i < indptr[r + 1]; ++i) {
+      if (indices[i] == c) {
+        out[tx] = i;
+        break;
+      }
+    }
+    rpos += row_stride;
+    cpos += col_stride;
+    tx += stride_x;
+  }
+}
+
+template <DLDeviceType XPU, typename IdType>
+bool CSRIsNonZero(CSRMatrix csr, int64_t row, int64_t col) {
+  CHECK(row >= 0 && row < csr.num_rows) << "Invalid row index: " << row;
```

Comment:
Maybe we add check for row and col < csr.dtype for int32

Reply:
Added check in the constructor of CSRMatrix and COOMatrix